File: backend/src/main/java/amos/group3/gitmodconfig_backend/models/CreateRepositoryModel.java
package amos.group3.gitmodconfig_backend.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
public class CreateRepositoryModel {
    private String name;

    private String owner;

    private SubmoduleModel[] submodules;

    private boolean auto_init;

    private boolean contentRepository;

    private boolean repoAlreadyOnGithub;
}

File: frontend/src/app/models/CreateRepositoryModel.ts
import { Submodule } from "./Submodule";

export class CreateRepositoryModel {
  name: string;
  contentRepository: boolean;
  submodules: Submodule[];
  owner: string;
  repoAlreadyOnGithub: boolean;

  constructor(
    name: string,
    submodules: Submodule[],
    isContentRepository: boolean,
    repoAlreadyOnGithub?: boolean,
    owner?: string
  ) {
    this.name = name;
    this.submodules = submodules;
    this.contentRepository = isContentRepository;
    this.owner = owner;
    this.repoAlreadyOnGithub = repoAlreadyOnGithub;
  }
}
